From 1e78096d569eecbd3201c821cb3484328caf8789 Mon Sep 17 00:00:00 2001
From: Pavla Kratochvilova <pkratoch@redhat.com>
Date: Fri, 5 Apr 2024 14:43:04 +0200
Subject: [PATCH 2/2] remove --oldinstallonly: when no old installonly
 packages, exit with 0
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream commit: 87eb5a7a3561381b5ef5e70548f49288251300fc

If no old installonly packages are present, then the command succesfully
removed all of them and should exit with 0 and write the message to
stdout instead of stderr.

Resolves: https://issues.redhat.com/browse/RHEL-6424
Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 dnf/cli/commands/remove.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/dnf/cli/commands/remove.py b/dnf/cli/commands/remove.py
index af9622fd..32e78d6d 100644
--- a/dnf/cli/commands/remove.py
+++ b/dnf/cli/commands/remove.py
@@ -124,8 +124,7 @@ class RemoveCommand(commands.Command):
                 for pkg in instonly:
                     self.base.package_remove(pkg)
             else:
-                raise dnf.exceptions.Error(
-                    _('No old installonly packages found for removal.'))
+                logger.info(_('No old installonly packages found for removal.'))
             return
 
         # Remove groups.
-- 
2.45.0

